### DISCUSSION TITLE
Could not install osxphotos

### DISCUSSION THREAD

**Discussion by trader0007** [RhetTbull/osxphotos#1177] [General]
when i do pip install osxphotos, I get "Failed building wheel for xattr"

OS 13.4

Tried pip install latter, pip install pyproject, pip install pyproject.toml .  All of those were successful.  But still pip install osxphotos gives me this error

**Comment by RhetTbull** [comment]
Hmm, that's weird.  I just tired it in a clean virtual environment with python 3.11.4 and macOS 13.4 and it worked fine.

My guess is the pip being installed is the wrong python version.  Which python are you using?

Try this: `python3 -m pip install osxphotos`

**Reply by trader0007** [reply]
did not work.  getting same error.  tried pipx as well with no luck.  this problem is in my old intel based Mac.  succesfully installed in my new m2 mac laptop

**Reply by RhetTbull** [reply]
What's the result of `python --version` and `python3 --version`? I use an old 2013 MPB for development of osxphotos so it does work on older Macs

**Comment by trader0007** [comment]
both say 3.11.4

**Reply by RhetTbull** [reply]
The dependency that's failing is in osxmetadata which is a dependency of osxphotos (also written by me). xattr does have wheels (pre-built packages) for macOS for both ARM (Apple Silicon) and x86 (Intel) though I see the 11.0 (Big sur & later) version is [only compiled for ARM](https://pypi.org/project/xattr/0.10.1/#files) so maybe that's the issue? It should actually have to build the package just install the wheel. The error you're getting indicates that pip is trying to build the package from source and failing. That might be because you need the XCode command line developer tools installed to build.  You could try `xcode-select --install` to install these tools then try again. This shouldn't be necessary but my guess is pip can't find a wheel for your macOS version & architecture.

Alternatively (maybe try this before install the Xcode tools), is to download the wheel for xattr and install that.  Download [this](https://files.pythonhosted.org/packages/f8/9a/834e987c3545e8408a7dec183d42385d34e76d151e5ee3d8751b504a8e91/xattr-0.10.1-cp311-cp311-macosx_10_9_x86_64.whl) file then run `python3 -m pip install /path/to/xattr-0.10.1-cp311-cp311-macosx_10_9_x86_64.whl`

**Reply by trader0007** [reply]
not that i dont trust you :), but what is in that file you are suggesting I download ?

**Reply by RhetTbull** [reply]
It's a dependency of osxmetadata, which is a dependency of osxphotos. The download is from the pypi.org official python package index. Here's the link to the pypi page for the file (doesn't download anything): [https://pypi.org/project/xattr/](https://pypi.org/project/xattr/). I'm just curious if you install the package manually if it'll work. I've not run into this issue before so just looking for options to get around the blocker you've encountered. You shouldn't have to download this manually but the normal way is not working.

**Reply by RhetTbull** [reply]
@trader0007 were you able to get osxphotos installed?
